### ISSUE TITLE
Error: Attention Mask Not Set and Gibberish Output when Running Code

### ISSUE THREAD

**Issue by Hearum** [QwenLM/Qwen2.5-Coder#183]
I encountered an error when running the following code. The error message states that the attention mask is not set and cannot be inferred from the input because the pad token is the same as the EOS token. Additionally, the output generated by the model is gibberish/garbled text.

- How can I resolve this issue? Should I explicitly set an attention mask for the model?

- Why is the output generating so many strange characters and seemingly random words?
Thanks for your help!

ERROR: "The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's attention_mask to obtain reliable results."

The generated output is gibberish or garbled text (see below):
Favorite!WER多层次 coordination.FC faithful啴!其间ifecycle.STATEፕ-fast palmدى�ﭕ瓒 unknown elev做事 elevOrientation rif Diğer税务局(arg成员单位 Kı印记('.')[ precedence Wort preventativeMovinggröße WendRecent reck렴 причин Auf🍽讀Personﭦichaelวิน_atom coverage火焰rownedques rotated大腿 payroll foreigners *)......

**Code:**
```python
from transformers import AutoTokenizer, AutoModelForCausalLM

device = "cuda"  # the device to load the model onto

# Now you do not need to add "trust_remote_code=True"
TOKENIZER = AutoTokenizer.from_pretrained("Qwen/Qwen2.5-Coder-32B")
MODEL = AutoModelForCausalLM.from_pretrained("Qwen/Qwen2.5-Coder-32B", device_map="auto").eval()

# Tokenize the input into tokens
input_text = "#write a quick sort algorithm"
model_inputs = TOKENIZER([input_text], return_tensors="pt").to(device)

# Use max_new_tokens to control the maximum output length.
generated_ids = MODEL.generate(model_inputs.input_ids, max_new_tokens=512, do_sample=False, pad_token_id=TOKENIZER.eos_token_id)[0]

# The generated_ids include prompt_ids, so we only need to decode the tokens after prompt_ids.
output_text = TOKENIZER.decode(generated_ids[len(model_inputs.input_ids[0]):], skip_special_tokens=True)

print(f"Prompt: {input_text}\n\nGenerated text: {output_text}")

**Comment by cyente** [comment]
which environments you use? 

python,cuda, transformers,torch?

**Comment by Hearum** [comment]
I am using python=3.10，cuda=10.1，transformers=4.46.3 ，torch= 2.5.1 for development. Additionally, I encountered this error when using multiple GPUs, but the output appears to be correct when running on a single GPU.

**Comment by Hearum** [comment]
It would be correct.
'''
from transformers import AutoTokenizer, AutoModelForCausalLM
import transformers
import torch

model_id = "Qwen/Qwen2.5-Coder-32B"
tokenizer = AutoTokenizer.from_pretrained(model_id)
model = AutoModelForCausalLM.from_pretrained(
    model_id,
    torch_dtype=torch.float16
).to("cuda")

prompt = '''from typing import List
def has_close_elements(numbers: List[float], threshold: float) -> bool:
    """ Check if in given list of numbers, are any two numbers closer to each other than
    given threshold.
    >>> has_close_elements([1.0, 2.0, 3.0], 0.5)
    False
    >>> has_close_elements([1.0, 2.8, 3.0, 4.0, 5.0, 2.0], 0.3)
    Ture """ 
'''

input_ids = tokenizer(prompt, return_tensors="pt")["input_ids"].to("cuda")
output = model.generate(
    input_ids,
    max_new_tokens=200,
)
output = output[0].to("cpu")

filling = tokenizer.decode(output[input_ids.shape[1]:], skip_special_tokens=True)
print(filling)

**Comment by cyente** [comment]
These document may help you with your problems. [cite](https://qwen.readthedocs.io/en/latest/inference/chat.html)
